eunkyung/app.js

```diff
@@ -34,25 +34,141 @@ app.get('/ping', (req, res) => {
+        `INSERT INTO posts (
+            title,
+            content,
+            post_image_url,
+            user_id
+        )
+        VALUES (
+            ?,
+            ?,
+            ?,
+            ?
+        );`,
+        [title, content, post_image_url, user_id]
+    )
+    res.status(201).json({ message: 'postCreated' })
+})
+
+app.get('/posts', async (req, res) => {
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        posts.user_id as postingId,
+        posts.post_image_url as postingImageUrl,
+        posts.content as postingContent 
+        FROM users INNER JOIN posts ON users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+
+app.get('/posts/:userId', async (req, res) => {
+    const { userId } = req.params;
+
+    await mysqlDataSource.query(
+        `SELECT
+        users.id as userId,
+        users.profile_image as userprofileImage,
+        pp.postings
+        FROM users
+        LEFT JOIN (
+            SELECT 
+            user_id,
+            JSON_ARRAYAGG(
+                JSON_OBJECT(
+                    'postingId', posts.id,
+                    'postingImageUrl', posts.post_image_url,
+                    'postingContent', posts.content
+                ) 
+            ) 
+        as postings
+        FROM posts 
+        GROUP BY user_id  
+        ) pp ON users.id = pp.user_id
+            WHERE users.id = ${userId};`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.patch('/edit/:userId/:postId', async (req, res) => {
+    const { userId, postId } = req.params;
+    const { postingContent } = req.body
+
+    await mysqlDataSource.query(
+        `UPDATE posts 
+        SET 
+        posts.content = ? 
+        WHERE posts.user_id = ${userId} AND posts.id = ${postId}
+        `,
+        [postingContent])
+
+
+    await mysqlDataSource.query(
+        `SELECT users.id as userId, users.name as userName, posts.id as postingId, posts.title as postingTitle, posts.content as postingContent FROM users INNER JOIN posts ON users.id = ${userId} AND posts.id = ${postId} WHERE users.id = posts.user_id`
+        , (err, rows) => {
+            res.status(201).json({ data: rows })
+        }
+    )
+})
+
+app.delete('/delete/:postId', async (req, res) => {
+    const { postId } = req.params;
+
+    await mysqlDataSource.query(
+        `DELETE FROM posts WHERE posts.id = ${postId}`
+    )
+    res.status(201).json({ message: 'postingDeleted' })
+})
+
+app.post('/like', async (req, res) => {
+    const { userId, postId } = req.body
+    await mysqlDataSource.query(
+        `INSERT INTO likes (
+            user_id,
+            post_id
+        )
+        VALUES (
+            ?,
+            ?
+        )`,
+        [userId, postId])
+    res.status(201).json({ message: 'likeCreated' })
+})
```

Comment:
👍 깔끔하군요! 포매팅 한번 신경써주면 좋을 것 같습니다.

```suggestion
app.post('/like', async (req, res) => {
    const { userId, postId } = req.body
    await mysqlDataSource.query(
        `INSERT INTO likes (
            user_id,
            post_id
        ) VALUES (
            ?,
            ?
        )`, [userId, postId])
    res.status(201).json({ message: 'likeCreated' })
})
```